### PR TITLE
Follow SQLAlchemy dialect attribute convention

### LINKED ISSUE #250: Include a `dialect` attribute to better follow common SqlAlchemy conventions relied upon by SqlAlchemy libraries
**Is your feature request related to a problem? Please describe.**
Other SqlAlchemy dialects define a `dialect` attribute, including but not limited to:
* [base.dialect = dialect = snowdialect.dialect](https://github.com/snowflakedb/snowflake-sqlalchemy/blob/38b8082f8ebf4106b6c9465996032b24806ad048/__init__.py#L62)
* [base.dialect = dialect = mysqldb.dialect](https://github.com/sqlalchemy/sqlalchemy/blob/76b506ed51e31b922014a30de2a5952d1a6ad891/lib/sqlalchemy/dialects/mysql/__init__.py#L60)
* [base.dialect = dialect = psycopg2.dialect](https://github.com/sqlalchemy/sqlalchemy/blob/76b506ed51e31b922014a30de2a5952d1a6ad891/lib/sqlalchemy/dialects/postgresql/__init__.py#L65)

Some libraries assume that this `dialect` attribute is defined, and without this the libraries work for every other dialect except for `BigQuery`. For example, the popular [Great Expectations library](https://github.com/great-expectations/great_expectations) commonly assumes this convention (e.g. see [this search](https://github.com/great-expectations/great_expectations/search?q=dialect.name) which shows some but not all usages). In the case of Great Expectations, BigQuery not following this convention currently results in multiple bugs that only affect BigQuery and no other dialects: https://github.com/great-expectations/great_expectations/issues/3131, https://github.com/great-expectations/great_expectations/issues/2978, and https://github.com/great-expectations/great_expectations/issues/2959.

 **Describe the solution you'd like**
BigQuery should follow the same conventions as other dialects so libraries using SqlAlchemy support BigQuery without special BigQuery specific code.

 **Describe alternatives you've considered**
Writing custom code for BigQuery when used with SqlAlchemy instead of relying on conventions used by all other dialects, however this is harder to maintain and will probably result in future bugs for multiple libraries that only affect BigQuery and not other dialects.

 **Additional context**
Please see proposed fix at Follow SQLAlchemy dialect attribute convention #251.

### PR DESCRIPTION
**Pull request by jdimatteo**
Fixes #250

### CODE REVIEW

File: sqlalchemy_bigquery/__init__.py [hunk 1]
```diff
@@ -41,6 +41,9 @@
     BIGNUMERIC,
 )
 
+# Define a "dialect" attribute to follow the common SqlAlchemy convention assumed by some libraries that leverage SqlAlchemy.
+base.dialect = dialect = BigQueryDialect
```
Comment (jdimatteo) [inline sqlalchemy_bigquery/__init__.py:45]: If you'd like me to write a test, please let me know what file I should add the test to.

File: sqlalchemy_bigquery/__init__.py [hunk 2]
```diff
@@ -41,6 +41,9 @@
     BIGNUMERIC,
 )
 
+# Define a "dialect" attribute to follow the common SqlAlchemy convention assumed by some libraries that leverage SqlAlchemy.
```
Comment (jdimatteo) [inline sqlalchemy_bigquery/__init__.py:44]: Other SqlAlchemy dialects define a `dialect` attribute, including but not limited to:
* [base.dialect = dialect = snowdialect.dialect](https://github.com/snowflakedb/snowflake-sqlalchemy/blob/38b8082f8ebf4106b6c9465996032b24806ad048/__init__.py#L62)
* [base.dialect = dialect = mysqldb.dialect](https://github.com/sqlalchemy/sqlalchemy/blob/76b506ed51e31b922014a30de2a5952d1a6ad891/lib/sqlalchemy/dialects/mysql/__init__.py#L60)
* [base.dialect = dialect = psycopg2.dialect](https://github.com/sqlalchemy/sqlalchemy/blob/76b506ed51e31b922014a30de2a5952d1a6ad891/lib/sqlalchemy/dialects/postgresql/__init__.py#L65)

See additional details on why this matters at #250.

Comment (jimfulton) [reply sqlalchemy_bigquery/__init__.py:44]: Interesting.  I wasn't aware of this.  Thanks!

This is a weird way to spell this. Why not define `dialect` in `base` and import it? :shrug:

File: sqlalchemy_bigquery/__init__.py (hunk 1 above)
Comment (jimfulton) [reply sqlalchemy_bigquery/__init__.py:45]: Thanks! I'd put this in `tests/unit/test_sqlalchemy_bigquery.py`.

**Comment by jimfulton** [comment]
I'll add the test...

**Comment by jimfulton** [comment]
> I'll add the test...

Well, no, I can't do that without creating a new PR.

**Review by jimfulton — CHANGES_REQUESTED** [review]
Please add a test in `tests/unit/test_sqlalchemy_bigquery.py`.

It can be as simple as checking that the dialect is present as the `dialect` attribute in `base` and in the package. Should take 5 minutes to write. :)

**Comment by jimfulton** [comment]
Superseded by #291

**Comment by jdimatteo** [comment]
> Please add a test in `tests/unit/test_sqlalchemy_bigquery.py`.
> 
> It can be as simple as checking that the dialect is present as the `dialect` attribute in `base` and in the package. Should take 5 minutes to write. :)

Sorry I didn't do this sooner -- thanks for fixing this with https://github.com/googleapis/python-bigquery-sqlalchemy/pull/291/files!

**Pull request closed without merging**